README.md

```diff
@@ -1,55 +1,35 @@
-# Necessary changes
+# Fleet HTTP client
 
-- timestamp for Message model needs to be of int64_t type, otherwise it overflows
-- Payload_data needs m_Json to store data
+A simple wrapper for a c++ rest API client generated by the [OpenAPI Generator](https://openapi-generator.tech). Simplifies functions used to call endpoints of the API. Also limits the amount of requests in case of an error which may cause the API to not wait for new statuses/commands. 
 
-# C++ API client
+### Request frequency limit
 
-HTTP-based API for Fleet Protocol v2 serving for communication between the External Server and the end users.
+Is only used on getCommands and getStatuses functions. Each time these functions get called, the current time gets added to a container which is then analyzed to find out whether there are too many requests. This needs to be configured by 4 parameters:
 
-## Overview
-This API client was generated by the [OpenAPI Generator](https://openapi-generator.tech) project. By using the [OpenAPI spec](https://openapis.org) from a remote server, you can easily generate an API client.
+- `maxRequestsThresholdCount` max amount of allowed requests in the specified time period
+- `maxRequestsThresholdPeriodMs` time period in which the amount of requests is checked
+- `delayAfterThresholdReachedMs` time to sleep for in ms when threshold is initially reached
+- `retryRequestsDelayMs` delay in ms between requests until request rate is no longer over the threshold
 
-- API version: 2.0.0
-- Package version: 
-- Build package: org.openapitools.codegen.languages.CppRestSdkClientCodegen
+After the initial threshold delay ends, the time container is emptied and subsequent requests are delayed for a shorter time. If the threshold is no longer reached afterwards, the next requests work without any delay as originally.
 
-- API namespace: org.openapitools.client.api
-- Model namespace: org.openapitools.client.model
+## Requirements
 
-## Installation
+- cpprestsdk: `sudo apt-get install libcpprest-dev`
 
-### Prerequisites
+## Generating client code
 
-Install [cpprestsdk](https://github.com/Microsoft/cpprestsdk).
+To regenerate the Cpp-rest-openapi-client lib, use the `regen.sh` script. It uses th openapi.yaml file as an input. The most up-to-date version can be found in the [fleet-protocol-http-api project](https://github.com/bringauto/fleet-protocol-http-api).
```

Comment:
What are the dependencies to run the script?

Reply:
mentioned requirements